### ISSUE TITLE
running test does not stop and test results not showing

### ISSUE THREAD

**Issue by antonblu** [formulahendry/vscode-dotnet-test-explorer#120]
Hello,

I was wondering if you can help. I have VS Code 1.26.1. When I run any of my C#  tests on xunit with selenium using "net core test explorer 0.5.0", the test progress icon right to the left of the running test just gives spinning, and never stops even though the test finished(I can see it finished as the browser automatically runs). Also, aside from the test progress icon just spinning forever, I also do not see anymore the test results in the terminal window. Previously, I would see either a success or fail(with details on which line the fail happened). Do I have to go back to a previous visual studio code version ?? or anything you can suggest ??

My tests that are in the following format used to work previously till July 2018 at least:

using System;
using Xunit;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using System.Linq;
using System.Diagnostics;
using OpenQA.Selenium.Support.UI;
using SeleniumExtras.WaitHelpers;

namespace xYZTests
{
    public class TestLibrary : IDisposable
    {
               variables etc
               driver etc

        public TestLibrary()
        {
            variables initialized here 

        }
        
        [Theory]
        [InlineData("variable data")]
        public void test1(variable names)
        {
         selenium automation statements here
        }
        
       [Theory]
        [InlineData("variable data")]
        public void test2(variable names)
        {
         selenium automation statements here
        }

[Theory]
        [InlineData("variable data")]
        public void test3(variable names)
        {
         selenium automation statements here
        }
  }

Thanks for your help.
Tonio

**Comment by stefanforsberg** [comment]
Hello, this behaviour usually indicates that vscode can't read from the folder the results files are placed in. The default value is the os temp folder which may be restricted depending on company policies etc. Try at set another path via the setting pathForResultFile.

Regarding test output this was moved to the output terminal, see https://github.com/formulahendry/vscode-dotnet-test-explorer#moved-test-output-to-output-terminal

**Comment by antonblu** [comment]
Hello Stefan,

I tried the "pathForResultFile" fix which I assume I put in the VS Code settings under 'Preferences' ?? but it did not work -unless this fix is placed in a different area ?
Next, I asked our senior developer and the following fix worked !! as follows :
my series of xunit test cases ends with the following last function to close the running automation of the browser:
public void Dispose()
        {
             sel._CloseDriver();
             //kill process chromedriver.exe
                 Process[] proc = Process.GetProcessesByName("ChromeDriver");
                 foreach (var p in proc)
                 {
                    p.Kill();
                    
                }
        }
However, because I wanted to demo the automation, I remmed the above code because I wanted to leave the last browser window open  and also or me to see the beauty of my automation as it came to a full stop :)
Anyway, this was the last thing I did and forgot about it because I was manually  testing another project.
When I came back to this automated test and noticed the above mentioned issue, the senior developer suggested the following fix as a compromise and it worked.. I unremmed the last function (shown above) and added a wait of whatever seconds you want e.g. sel.Wait(5 * 1000);   so at least I can see  the glorious last window before it closes and that fixed it ! 
public void Dispose()
        {
            
             sel.Wait(5 * 1000);
             sel._CloseDriver();
             //kill process chromedriver.exe
                 Process[] proc = Process.GetProcessesByName("ChromeDriver");
                 foreach (var p in proc)
                 {
                    p.Kill();
                    
                }

In other words, if the last closing function is not included, the test progress icon will keep spinning and you wont see the pass or fail results.

**Comment by stefanforsberg** [comment]
@antonblu Hello!

Ahhh ok I see, we use the exit of the dotnet test process as a trigger that the test run is completed and we can start parsing the results. In this case when the chromedriver process is never killed causing the the dotnet test process to never complete and hence we don't show any test results.

Happy you found a way around it and thank you for your kind words =)
